R/verb-fill.R

```diff
@@ -0,0 +1,169 @@
+# ignore nulls
```

Comment:
I think this would be easier for me to review if you could start with a high-level description of how the translation will work (for most databases). Here I think the key idea is using `LAST_VALUE` + `IGNORE NULLS` + the default frame (which goes from the first row in the group to the "current" row).

Reply:
I explained both strategies in the respective function. I hope this helps in understanding it better.

Reply:
Looks good — can you please move this comment down close to `fill0` so it's closer to the implementations?